From: Tejun Heo <htejun@gmail.com>
Date: Mon, 6 Aug 2007 17:43:27 +0000 (+0900)
Subject: ata_piix: update map 10b for ich8m
X-Git-Tag: v2.6.23-rc4~131^2~5
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=ac2b04371fffd964b0d1c3369a9972bed7a5c5d9

ata_piix: update map 10b for ich8m

Fix map entry 10b for ich8.  It's [P0 P2 IDE IDE] like ich6 / ich6m.

Signed-off-by: Tejun Heo <htejun@gmail.com>
Cc: <Kristen Carlson Accardi> kristen.c.accardi@intel.com
Signed-off-by: Jeff Garzik <jeff@garzik.org>
---

Backported to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-2.6-2.6.18.dfsg.1.orig/drivers/scsi/ata_piix.c linux-2.6-2.6.18.dfsg.1/drivers/scsi/ata_piix.c
--- linux-2.6-2.6.18.dfsg.1.orig/drivers/scsi/ata_piix.c	2007-09-12 18:47:41.000000000 -0600
+++ linux-2.6-2.6.18.dfsg.1/drivers/scsi/ata_piix.c	2007-09-12 18:50:38.000000000 -0600
@@ -373,9 +373,9 @@ static const struct piix_map_db ich8_map
 	.present_shift = 8,
 	.map = {
 		/* PM   PS   SM   SS       MAP */
-		{  P0,  NA,  P1,  NA }, /* 00b (hardwired) */
+		{  P0,  P2,  P1,  P3 }, /* 00b (hardwired when in AHCI) */
 		{  RV,  RV,  RV,  RV },
-		{  RV,  RV,  RV,  RV }, /* 10b (never) */
+		{  P0,  P2, IDE, IDE }, /* 10b (IDE mode) */
 		{  RV,  RV,  RV,  RV },
 	},
 };
